routes/errortracker.js

```diff
@@ -0,0 +1,250 @@
+/**
+ * Created by rigwanzojunior on 6/5/17.
+ *objects used
+ *Objects are identical to the objects used in previous errortracker.go
+ * @typedef errorRequestMeta {
+	HTTPReferrer: string,
+	HTTPUserAgent: string
+    }
+  @typedef errorRequest {
+	URL:  string,
+	Method: string,
+	Meta: errorRequestMeta
+  }
+  @typedef errorEvent {
+	Application: string,
+	AppID; string,
+	Environment: string,
+	Version: string,
+	Message: string,
+	Exception: string,
+	Request: *errorRequest,
+	Filename: string,
+	Line: int32,
+	Classname: string,
+	Function: string,
+	Severity: string
+   }
+ * Handle error requests from clients and log them.
+ */
+// AppEngine project ID
+const projectId ='';
+const logName = 'javascript.errors';
+const express = require('express');
+const router = express.Router();
+const logging = require('@google-cloud/logging');
+const winston = require('winston');
+const statusCodes = require('http-status-codes');
+const Math = require('./Math');
+const url = require('url');
+/**
+ * errorLevels
+ * @enum {string}
+ */
+const errorLevels = {
+    Default: 'Default',
+    Debug: 'Debug',
+    Info: 'Info',
+    Notice: 'Notice',
+    Warning: 'Warning',
+    Error: 'Error',
+    Critical: 'Critical',
+    Alert: 'Alert',
+    Emergency: 'Emergency',
+};
+// params in http request
+let params = {};
+
+/**
+ * Filter exceptions in an array to prevent them from being logged
+ * @param errorEvent
+ * @returns {boolean}
+ */
+function isFilteredMessageOrException(errorEvent) {
+    let filteredMessageOrException = ['stop_youtube',
+        'null%20is%20not%20an%20object%20(evaluating%20%27elt.parentNode%27)'];
+    filteredMessageOrException.some(function filter(msg) {
+        if (errorEvent.Message.toString().includes(msg) || errorEvent.Exception.toString().includes(msg)) {
+            return true;
+        }
+    });
+    return false;
+}
+
+/**
+ * Function that handles errors that come from an attempt to write to the logs
+ * @param err error
+ * @param res http response object
+ * @param req http request object
+ */
+function logWritingError(err, res,req) {
+    if (err) {
+        res.status(statusCodes.INTERNAL_SERVER_ERROR).send({error: 'Cannot write to Google Cloud Logging'});
+        winston.error(projectId, 'Cannot write to Google Cloud Logging: '+url.parse(req.url, true).query['v'], err);
+    }
+}
+
+/**
+ *extract params in GET request from query and fill errorEvent object
+ * @param req
+ * @param res
+ * @param next
+ */
+function getHandler(req, res, next) {
+    params = url.parse(req.url, true).query;
+    let referer = req.get('Referer');
+    let resource = {
+        type: 'compute.googleapis.com',
+        labels: {
+            'compute.googleapis.com/resource_type': 'logger',
+            'compute.googleapis.com/resource_id': 'errors',
+        },
+    };
+    let line = params.l;
+    let errorType = 'default';
+    let isUserError = false;
+    if (params.a === 1) {
```

Comment:
So `a` may be numeric, but it's type will be a string. This is because the url parser takes in a string (`"/r?a=1"`), and outputs a object (for the `query`) of type `Object<string, string>`. All of these `=== 1` will fail then, because `'1' !== 1`